spec/index.bs

```diff
@@ -1493,6 +1497,103 @@ interface GPUTextureUsage {
+        1. Let |d| = 1.
+
+      1. Else if the {{GPUTextureDescriptor/dimension}} is "2d":
+
+        1. Let |w| = {{GPUTextureDescriptor/size}}.[=Extent3D/width=].
+        1. Let |h| = {{GPUTextureDescriptor/size}}.[=Extent3D/height=].
+        1. Let |d| = 1.
+
+      1. Else (the {{GPUTextureDescriptor/dimension}} is "3d"):
+
+        1. Let |w| = {{GPUTextureDescriptor/size}}.[=Extent3D/width=].
+        1. Let |h| = {{GPUTextureDescriptor/size}}.[=Extent3D/height=].
+        1. Let |d| = {{GPUTextureDescriptor/size}}.[=Extent3D/depth=].
+
+      1. Let |m| = the maximum value of |w|, |h|, and |d|.
+      1. Return one plus the greatest integral value of |x| for which 2^|x| <= |m|.
+
+    </dfn>
+  </div>
+
+  <div algorithm class=validusage>
+    <dfn abstract-op>createTexture Valid Usage</dfn>
+      Given a {{GPUDevice}} |this| and a {{GPUTextureDescriptor}} |descriptor|
+      the following validation rules apply:
+
+      1. |this| must be a [=valid=] {{GPUDevice}}.
+      1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be nonzero.
+      1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be nonzero.
+      1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be nonzero.
+      1. |descriptor|.{{GPUTextureDescriptor/mipLevelCount}} must be nonzero.
+      1. |descriptor|.{{GPUTextureDescriptor/sampleCount}} must be nonzero.
+      1. If |descriptor|.{{GPUTextureDescriptor/dimension}} is "1d":
+
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be less than or equal to 8192.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be less than or equal to 2048.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be 1.
+        1. |descriptor|.{{GPUTextureDescriptor/sampleCount}} must be 1.
+
+      1. Else if |descriptor|.{{GPUTextureDescriptor/dimension}} is "2d":
+
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be less than or equal to 8192.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be less than or equal to 8192.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be less than or equal to 2048.
+
+      1. Else (|descriptor|.{{GPUTextureDescriptor/dimension}} is "3d"):
+
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be less than or equal to 2048.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be less than or equal to 2048.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be less than or equal to 2048.
+        1. |descriptor|.{{GPUTextureDescriptor/sampleCount}} must be 1.
+
+      1. If |descriptor|.{{GPUTextureDescriptor/sampleCount}} > 1:
+
+        1. |descriptor|.{{GPUTextureDescriptor/mipLevelCount}} must be 1.
+        1. |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depth=] must be 1.
```

Comment:
we should also say that the dimension has to be "2d", i.e. disallow 1D multisampled textures (for now)

Reply:
It is done in that dimension's list of validation checks, but I think it would be good to have all sample count constraints in the same place.

Reply:
I couldn't get Bikeshed to give me nested "if"s :(

Reply:
I think it works if you add enough indentation. But sometimes 2 spaces is enough while other times you need 4 spaces. It's a bit weird.

Reply:
Generally it requires 4 spaces; I think it usually treats 2 spaces similarly to 0 spaces.